admiral/pkg/controller/admiral/vertex.go

```diff
@@ -0,0 +1,268 @@
+package admiral
+
+import (
+	"context"
+	"fmt"
+	"github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	numaflow "github.com/numaproj/numaflow/pkg/client/clientset/versioned"
+	v1alpha12 "github.com/numaproj/numaflow/pkg/client/informers/externalversions/numaflow/v1alpha1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sync"
+	"time"
+
+	"github.com/istio-ecosystem/admiral/admiral/pkg/client/loader"
+	"github.com/istio-ecosystem/admiral/admiral/pkg/controller/common"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/rest"
+
+	"k8s.io/client-go/tools/cache"
+)
+
+//Vertex controller discovers vertexs as mesh clients (its assumed that k8s Vertex doesn't have any ingress communication)
+
+type VertexController struct {
+	NumaflowClient   numaflow.Interface
+	VertexHandler  ClientDiscoveryHandler
+	informer    cache.SharedIndexInformer
+	Cache       *vertexCache
+}
+
+type VertexEntry struct {
+	Identity string
+	Vertices map[string]*K8sObject
+}
+
+type vertexCache struct {
+	//map of dependencies key=identity value array of onboarded identities
+	cache map[string]*VertexEntry
+	mutex *sync.Mutex
+}
+
+func NewVertexCache() *vertexCache {
+	return &vertexCache{
+		cache: make(map[string]*VertexEntry),
+		mutex: &sync.Mutex{},
+	}
+}
+
+
+func (p *vertexCache) getK8sObjectFromVertex(vertex v1alpha1.Vertex) *K8sObject{
```

Comment:
We probably don't need a pointer receiver here

Reply:
@kpharasi You mean to K8sObject can be returned as value and not reference?

Reply:
I meant we can directly call this function rather than using the `vertexCache` object to call it  as we are not using any field from the `vertexCache` struct.